src/Microsoft.Azure.WebJobs.Extensions.Kafka/Trigger/KafkaTriggerAttributeBindingProvider.cs

```diff
@@ -100,10 +100,28 @@ private KafkaListenerConfiguration CreateConsumerConfiguration(KafkaTriggerAttri
             {
                 consumerConfig.SaslPassword = this.config.ResolveSecureSetting(nameResolver, attribute.Password);
                 consumerConfig.SaslUsername = this.config.ResolveSecureSetting(nameResolver, attribute.Username);
-                consumerConfig.SslKeyLocation = this.config.ResolveSecureSetting(nameResolver, attribute.SslKeyLocation);
                 consumerConfig.SslKeyPassword = this.config.ResolveSecureSetting(nameResolver, attribute.SslKeyPassword);
-                consumerConfig.SslCertificateLocation = this.config.ResolveSecureSetting(nameResolver, attribute.SslCertificateLocation);
-                consumerConfig.SslCaLocation = this.config.ResolveSecureSetting(nameResolver, attribute.SslCaLocation);
+
+                var sslKeyLocation = this.config.ResolveSecureSetting(nameResolver, attribute.SslKeyLocation);
+                if (!AzureFunctionsFileHelper.TryGetValidFilePath(sslKeyLocation, out var resolvedSslKeyLocation))
+                {
+                    resolvedSslKeyLocation = sslKeyLocation;
+                }
+
+                var sslCertificateLocation = this.config.ResolveSecureSetting(nameResolver, attribute.SslCertificateLocation);
+                if (!AzureFunctionsFileHelper.TryGetValidFilePath(sslCertificateLocation, out var resolvedSslCertificationLocation))
+                {
+                    resolvedSslCertificationLocation = sslCertificateLocation;
+                }
```

Comment:
Can the if condition be avoided by using the same sslCertificateLocation variable as the out parameter as well? For example -  AzureFunctionsFileHelper.TryGetValidFilePath(sslCertificateLocation, out sslCertificateLocation)

Reply:
I don't think this would make sense. If you check the TryGetValidFilePath() function, it sets out parameter to null.

Reply:
Yes, it does set the out parameter to null. But it reassigns the value if the file path exists and makes it null if the file path does not exist. It's a choice about how we want to handle the case where the file path does not exist. Do we want to pass a file path even when we know it does not exist?

Reply:
Your point seems valid. I haven't removed the if condition but have thrown error if the path is invalid.